metals/src/main/scala/scala/meta/internal/metals/Doctor.scala

```diff
@@ -69,10 +70,11 @@ final class Doctor(
       clientCommand: Command,
       onServer: MetalsHttpServer => Unit
   ): Unit = {
-    if (config.executeClientCommand.isOn) {
+    if (config.executeClientCommand.isOn || clientExperimentalCapabilities.executeClientCommandProvider) {
       val output =
-        if (config.doctorFormat.isHtml) buildTargetsHtml()
-        else buildTargetsJson()
+        if (config.doctorFormat.isJson || clientExperimentalCapabilities.doctorFormatIsJson)
```

Comment:
I flipped the order of these since both default to html. So, I assume that if either is set to Json, it was intentionally set, and we then go with that.